content/articles/line-detection-using-hough-transform-in-matlab/index.md

```diff
@@ -0,0 +1,149 @@
+
+![Expression using rho, and theta](/engineering-education/line-detection-using-hough-transform-in-matlab/image-two.png)
+
+Conceptually, hough transform is the mapping of image coordinate in the x and y into the parametric coordinate $\theta, \rho$
+
+### Using hough transform to detect a line
+Suppose an image consists of a line as shown below;
+
+![Your line](/engineering-education/line-detection-using-hough-transform-in-matlab/image-three.png)
+
+A line is made up of multiple points. Similarly, multiple lines can pass through the same point. We visualize multiple lines in a hough transform plane where the parameters $\theta$ are on the x-axis and $\rho$ on the y-axis. Every point on the image is transformed into a sinusoid.
+
+![Image in the hough transform plane](/engineering-education/line-detection-using-hough-transform-in-matlab/image-four.png)
+
+It makes sense because many lines may pass through a given point in an image plane. It translates into a sinusoid in the hough plane. If two or more of these lines coincide to form a line in the image plane, they will intersect in the hough transform plane. This intersection corresponding to the $\theta, \rho$ pair corresponds to the detected line.
+
+![Conciding lines](/engineering-education/line-detection-using-hough-transform-in-matlab/image-five.png)
+
+> Note that the number of sinusoids intersecting determined the strengths of a line. It is how hough transformation works.
+
+### Extracting line segments using hough transform in Matlab
+Suppose you have an image with a line and you want to extract the line segments; the algorithm is:
+1. Create a hough transform matrix using the hough function.
+2. Locate the peaks in the Hough transform matrix function. The peaks correspond to the intersection point in the parameter plane. Each of these peaks is a detected line.
+3. Use a hough transform function to map the peak back into the image plane. You then extract the line segments.
+
+Now, we use a simple image to implement the above algorithm. We have two binary images shown below;
+
+![Binary images](/engineering-education/line-detection-using-hough-transform-in-matlab/image-six.png)
+
+These images are binary because hough transform function works on binary images. Note that it is possible to plot these two images in Matlab. We will first plot these two images and extract line segments. We first plot the image with the two dots and extract lines from it before going to the second.
+To plot the image with the two dots, we use execute the code below;
```

Comment:
```suggestion
These images are binary because hough transform function works on binary images. Note that it is possible to plot these two images in Matlab. We will first plot these two images and extract line segments. We first plot the image with the two dots and extract lines from it before going to the second.

To plot the image with the two dots, we will execute the code below;

```